addons/crm/models/crm_lead.py

```diff
@@ -672,6 +672,12 @@ def create(self, vals_list):
         for vals in vals_list:
             if vals.get('website'):
                 vals['website'] = self.env['res.partner']._clean_website(vals['website'])
+
+            if not vals.get('lang_id') and vals.get('partner_id'):
+                partner_lang = self.env['res.partner'].browse(vals['partner_id']).lang
+                if partner_lang:
+                    vals['lang_id'] = self.env['res.lang']._lang_get(partner_lang).id
+
```

Comment:
shouldn't you do a compute method instead ? (like we do for name, contact_name, partner_name, function,...)